crypto/o_time.c

```diff
@@ -47,6 +47,35 @@ typedef struct _ile3 {          /* Copied from ILEDEF.H for Alpha   */
 # endif   /* alpha || ia64    */
 #endif    /* OPENSSL_SYS_VMS  */
 
+time_t OPENSSL_timegm(struct tm *tm)
+{
+    if (1) {
+#if defined(__USE_MISC)
+        return timegm(tm);
+    } else {
+#endif
+        /* Figures it out without mucking the environment */
+        time_t currtime, utc, local;
+        struct tm *timeinfo;
+        int offset;
+
+        time(&currtime);
+        timeinfo = gmtime(&currtime);
+        utc = mktime(timeinfo);
+
+        timeinfo = localtime(&currtime);
+        local = mktime(timeinfo);
+
+        offset = (int)difftime(utc, local);
+
+        /* Adjust for DST */
+        if (timeinfo->tm_isdst)
+            offset -= 3600;
```

Comment:
Unfortunately, there's no real way to know the amount. Do you have a concrete example?